src/NuGet.Core/NuGet.Protocol.Core.v3/HttpSource/HttpSource.cs

```diff
@@ -171,6 +172,19 @@ public HttpSource(PackageSource source, Func<Task<HttpHandlerResource>> messageH
             }
         }
 
+        public Task<HttpResponseMessage> GetAsync(Uri uri, MediaTypeWithQualityHeaderValue[] accept, ILogger log, CancellationToken token)
+        {
+            var request = new HttpRequestMessage(HttpMethod.Get, uri);
+            foreach (var a in accept)
+            {
+                request.Headers.Accept.Add(a);
+            }
+
+            Func<HttpRequestMessage> requestFactory = () => request;
```

Comment:
This should create a completely new request, see: https://github.com/NuGet/NuGet.Client/pull/378